library/bignum_mod.c

```diff
@@ -135,6 +135,16 @@ static int set_mont_const_square( const mbedtls_mpi_uint **X,
     return( ret );
 }
 
+static inline int mpi_core_is_zero( const mbedtls_mpi_uint *p, size_t limbs )
+{    
+    size_t zero_c;
+
+    for( zero_c = 0; zero_c < limbs; zero_c++ )
+        if ( p[zero_c] != (mbedtls_mpi_uint) 0 ) break;
```

Comment:
It seems a bit of a shame to introduce a non-constant time function like this here, when we could do something simple like

```
static inline int mpi_core_is_zero( const mbedtls_mpi_uint *A, size_t limbs )
{
    mbedtls_mpi_uint bits = 0;

    for( size_t i = 0; i < limbs; i++ )
        bits |= A[i];

    return (bits == 0) ? 1 : 0;
}
```

Reply:
I considered about constant time but that was is a tricky one. 

Do we need contant time at the expense on perfomance for an input checking utility function which is called multiple times in a normal flow?

Reply:
@yanesca Do you have an opinion on this? ^

Reply:
I too think that it is a low hanging fruit from the perspective of constant timeness. Still we would need to instrument the tests for that. Also, `return (bits == 0) ? 1 : 0;` introduces a branch so we would need to address that.

In summary, it is tempting to do it, but it is not as trivial as it seems and it is out of scope for now.